addons/transifex/models/models.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from configparser import ConfigParser
+from os.path import join as opj
+import os
+import werkzeug.urls
+
+import odoo
+from odoo import models
+
+class BaseModel(models.AbstractModel):
+    _inherit = 'base'
+
+    def get_field_translations(self, field_name, langs=None):
+        """ :return dict translations: [{lang: lang, source: val_en, value: val_lang, transifexUrl: transifex_url)] """
+        translations, context = super().get_field_translations(field_name, langs=langs)
+        self._get_transifex_url(translations)
+        return translations, context
+
+    def _get_transifex_url(self, translations):
+        """ Construct transifex URL based on the module on configuration """
+        self.ensure_one()
+        # .tx/config files contains the project reference
+        # using ini files like '[odoo-master.website_sale]'
+        external_id = self.get_external_id().get(self.id)
+        if not external_id:
+            return
+        module = external_id.split('.')[0]
+        if not module:
+            return
+
+        # e.g. 'https://www.transifex.com/odoo/'
+        base_url = self.env['ir.config_parameter'].sudo().get_param('transifex.project_url')
+
+        tx_config_file = ConfigParser()
+        tx_sections = []
+        for addon_path in odoo.addons.__path__:
+            tx_path = opj(addon_path, '.tx', 'config')
+            if os.path.isfile(tx_path):
+                tx_config_file.read(tx_path)
+                # first section is [main], after [odoo-11.sale]
+                tx_sections.extend(tx_config_file.sections()[1:])
+
+            # parent directory ad .tx/config is root directory in odoo/odoo
+            tx_path = opj(addon_path, os.pardir, '.tx', 'config')
+            if os.path.isfile(tx_path):
+                tx_config_file.read(tx_path)
+                tx_sections.extend(tx_config_file.sections()[1:])
+
+        if not (base_url and tx_sections):
+            return
+
+        base_url = base_url.rstrip('/')
+
+        project = None
+        for section in tx_sections:
+            tx_project, tx_mod = section.split('.')
+            if tx_mod == module:
+                project = tx_project
+                break
```

Comment:
Hi, @mart-e 

I hope I can avoid using the `for` loop for all sections to get the `project`.
Because now we only handle one field of one record of one module
Is it possible
1. Assume all sections' project names in one config file are the same.
2. Store the project name in somewhere(maybe another config file)?
3. get the project name and use `f'{project}.{module} in tx_config_file` to know if it is valid
Or
1. Store all distinct project names in somewhere(maybe another config file)?
2. Use a `for` loop for each project name and check if `f'{project}.{module} in tx_config_file`
Or
You may have a better idea 😃

Reply:
Why do you want to avoid the for loop? Storing in another config files seems quite cumbersome.

Reply:
the current code is O(n) where n is the number of modules in the config
check `f'{project}.{module}' in tx_config_file` seems like O(1) if we know all `project` candidates

Maybe we can put those project names in transifex_data.xml `ir.config_parameter` with `noupdate=False`?

Reply:
Going through a list of a few hundreds items in memory vs making a couple of database calls? Not sure it's worth the effort for something that is already negligeable.